bindings/java/src/main/java/org/apache/opendal/OperatorOutputStream.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.opendal;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.Arrays;
+
+public class OperatorOutputStream extends OutputStream {
+    private static class Writer extends NativeObject {
+        private Writer(long nativeHandle) {
+            super(nativeHandle);
+        }
+
+        @Override
+        protected void disposeInternal(long handle) {
+            disposeWriter(handle);
+        }
+    }
+
+    private static final int MAX_BYTES = 16384;
+
+    private final Writer writer;
+    private final byte[] bytes = new byte[MAX_BYTES];
+
+    private int offset = 0;
+
+    public OperatorOutputStream(BlockingOperator operator, String path) {
+        final long op = operator.nativeHandle;
+        this.writer = new Writer(constructWriter(op, path));
+    }
+
+    @Override
+    public void write(int b) throws IOException {
+        bytes[offset++] = (byte) b;
+        if (offset >= MAX_BYTES) {
+            flush();
```

Comment:
We don't need this, opendal core handling buffer by their own.

Reply:
We need it.

This sends the Java buffer to the Rust. Otherwise the Rust operator never receive the buffer. We should buffer it a bit in the Java side because if we send every byte, it would introduce expensive JNI call total cost.

Reply:
That is, they are different concept of buffering.